File: 3471-minimum-average-of-smallest-and-largest-elements/minimum-average-of-smallest-and-largest-elements.py
class Solution:
    def minimumAverage(self, nums: List[int]) -> float:
        min_ = float('inf')
        nums.sort()
        l = 0
        r = len(nums) - 1

        while l < r:
            avg = (nums[l] + nums[r])/ 2
            min_ = min(min_, avg)
            l += 1
            r -= 1

        return min_

File: 3471-minimum-average-of-smallest-and-largest-elements/minimum-average-of-smallest-and-largest-elements.java
class Solution {
    public double minimumAverage(int[] nums) {
        int l = 0;
        int r = nums.length - 1;
        double min = Double.MAX_VALUE;
        Arrays.sort(nums);
        while(l < r){
            double avg = (double) (nums[l] + nums[r]) / 2;
            min = Math.min(min, avg);
            l++;
            r--;
        }
        return min;
    }
}